src/Tokenizer/Analyzer/Analysis/DataProviderAnalysis.php

```diff
@@ -14,20 +14,30 @@
 
 namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
 
+use PhpCsFixer\Console\Application;
+use PhpCsFixer\Utils;
+
 final class DataProviderAnalysis
 {
     private string $name;
 
     private int $nameIndex;
 
-    /** @var array<int> */
+    /** @var list<int> */
     private array $usageIndices;
 
     /**
-     * @param array<int> $usageIndices
+     * @param list<int> $usageIndices
      */
     public function __construct(string $name, int $nameIndex, array $usageIndices)
     {
+        if (!array_is_list($usageIndices)) {
+            Utils::triggerDeprecation(new \InvalidArgumentException(sprintf(
+                'Parameter "usageIndices" should be of type 'list' (and not type 'array') and this will be enforced in version %d.0.',
```

Comment:
But technically `list` is an `array` and only the latter is a native type, that's why it can be confusing. So if "[we do not describe the difference between them anywhere](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7800#discussion_r1476871657)" then I believe it should be simplified to this:

```suggestion
                'Parameter "usageIndices" should be a list and this will be enforced in version %d.0.',
```

Assuming everyone knows what's the list, then there's no need to add "not an array" part.